src/operator/tensor/elemwise_binary_op-inl.h

```diff
@@ -450,6 +450,85 @@ void ElemwiseBinaryOp::DnsCsrDnsOp(mshadow::Stream<cpu> *s,
   });
 }
 
+/*!
+ * \brief Kernel for performing elemwise op between dense and rsp tensor
+ * \param i            global thread id
+ * \param req          type of request
+ * \param out          output array
+ * \param dns_data     data array of dense input
+ * \param rsp_data     data array of rsp input
+ * \param rsp_indices  indices array of rsp input
+ * \param num_rows     number of rows of both inputs
+ * \param nz_rows      number of non-zero rows of rsp tensor
+ * \param num_cols     number of columns of both inputs
+ */
+template<typename OP>
```

Comment:
Usually req should be templated to reduce the branches in kernel
https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/control_flow_op.h#L44